app/services/sign_in/access_token_audience_generator.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module SignIn
+  class AccessTokenAudienceGenerator
+    SHARED_SESSION_CLIENT_IDS_CACHE_KEY = 'sis_shared_sessions_client_ids'
+
+    def initialize(client_id:)
+      @client_id = client_id
+    end
+
+    def perform
+      generate_audience
+    end
+
+    private
+
+    attr_reader :client_id
+
+    def generate_audience
+      return shared_session_client_ids if client_id.in?(shared_session_client_ids)
+
+      [client_id]
+    end
+
+    def shared_session_client_ids
+      @shared_session_client_ids ||= Rails.cache.fetch(SHARED_SESSION_CLIENT_IDS_CACHE_KEY, expires_in: 24.hours) do
```

Comment:
I think we can prob just call Client Config directly here instead of making a cache since this will only be called when we're creating/refreshing a session

Reply:
Yeah I could go either way.  I benchmarked this on staging and the cache was 30% faster and I figured it wouldn't change that much. With mobile only it gets called ~300,000 times a day but it's not a huge difference when we're talking milliseconds

Reply:
I changed the cache expiration to 5 minutes